test/base/testing_brave_browser_process.h

```diff
@@ -0,0 +1,90 @@
+/* Copyright (c) 2021 The Brave Software Team. Distributed under the MPL2
+ * license. This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// An implementation of BraveBrowserProcess for unit tests that fails for most
+// services. By preventing creation of services, we reduce dependencies and
+// keep the profile clean. Clients of this class must handle the NULL return
+// value, however.
+
+#ifndef BRAVE_TEST_BASE_TESTING_BRAVE_BROWSER_PROCESS_H_
+#define BRAVE_TEST_BASE_TESTING_BRAVE_BROWSER_PROCESS_H_
+
+#include <stdint.h>
+
+#include <memory>
+#include <string>
+
+#include "brave/browser/brave_browser_process.h"
+
+namespace brave_shields {
+class AdBlockService;
+}
+
+class TestingBraveBrowserProcess : public BraveBrowserProcess {
```

Comment:
can we please also add a scoped util similar to `TestingBrowserProcessInitializer` ?

Reply:
I added it since it was pretty simple anyways, although I'm not sure if it will be useful for us. Just looking at the `TestingBrowserProcessInitializer` in cs.chromium.org, it's only used in two unit test classes (`ChildStatusCollectorTest` and `DeviceStatusCollectorTest`).